test/integration/retryable-writes/retryable_writes.spec.test.ts

```diff
@@ -63,16 +79,17 @@ describe('Legacy Retryable Writes Specs', function () {
         await ctx.client.close();
         ctx = {}; // reset context
       });
-
       for (const spec of suite.tests) {
-        // Step 2: Run the test
-        const mochaTest = it(spec.description, async () => await executeScenarioTest(spec, ctx));
-
-        // A pattern we don't need to repeat for unified tests
-        // In order to give the beforeEach hook access to the
-        // spec test so it can be responsible for skipping it
-        // and executeScenarioSetup
-        mochaTest.spec = spec;
+        if (!LEGACY_SKIP_TESTS_4_4_SHARDED.includes(spec.description)) {
```

Comment:
Unsure how to detect server type and version within this file (the `Context` given is empty and so is `this.configuration`). Because of this, I've skipped all of the legacy tests rather than only `serverVersion <= 4.4`  and `topologyType === 'Sharded'`.

Let me know your thoughts and suggestions!

Reply:
See my comment in `error.ts`, we shouldn't have to skip tests.  

But in general - `this.configuration` is only accessible in `before`/`after`/`test` functions, not in the `context`/`describe` blocks.  If you put skip logic into a `beforeEach`, you should be able to skip based on server version and topology.